org.scala-ide.sdt.core/src/org/scalaide/ui/internal/preferences/CallByNameParamCreationPreferencePage.scala

```diff
@@ -0,0 +1,44 @@
+package org.scalaide.ui.internal.preferences
+
+import org.eclipse.swt.widgets.Composite
+import org.eclipse.swt.widgets.Control
+import org.eclipse.core.runtime.preferences.AbstractPreferenceInitializer
+import org.scalaide.core.IScalaPlugin
+
+class CallByNameParamCreationPreferencePage extends BasicFieldEditorPreferencePage("Configure highlighting for the creation of call-by-name parameters") {
+  import CallByNameParamCreationPreferencePage._
+
+  override def createContents(parent: Composite): Control = {
+    val control = super.createContents(parent).asInstanceOf[Composite]
+    PrefPageUtils.mkLinkToAnnotationsPref(parent)(a => s"More options for highlighting for call-by-name parameters on the $a preference page.")
+    control
+  }
+
+  override def createFieldEditors() {
+    addBooleanFieldEditors(
+      P_ACTIVE -> "Enabled",
+      P_BOLD -> "Bold",
+      P_ITALIC -> "Italic",
+      P_FIRST_LINE_ONLY -> "Only highlight the first line")
+  }
+}
+
+object CallByNameParamCreationPreferencePage {
+  val BASE = "scala.tools.eclipse.ui.preferences.callByNameParamCreation"
```

Comment:
I prefer `CamelCase`, but it seems to me that most constants in Scala-IDE are `CAPITALIZED_WITH_UNDERSCORE` (see org.scalaide.ui.syntax.ScalaSyntaxClasses for a bunch of examples). I don't think that is a good idea to mix conventions.

As for inlining `BASE`: Here I'm also just doing things like they are done by other pages (like `org.scalaide.ui.internal.preferences.OrganizeImportsPreferencesPage`, `org.scalaide.ui.internal.preferences.EditorPreferencePage`, ...). I think in the end it is a matter of taste what you prefer, as both approaches have advantages and disadvantages that cannot be easily compared in an objective manner.